.github/workflows/python-checks.yml

```diff
@@ -88,18 +88,25 @@ jobs:
       - name: Install Python dependencies
```

Comment:
Nit: please update the PR title

Reply:
Also worth mentioning that this is used just for type hinting and does not have any runtime performance implications.

Reply:
Theoretically it could have a very slight performance impact with the `assert ... is not None` if Python's ran in debug mode. I don't think it's worth worrying about

Reply:
I meant the type hinting itself, not the assertions.

Reply:
But I agree those are negligible.